### ISSUE TITLE
Confusing output in FATES-SP history files

### ISSUE THREAD

**Issue by wwieder** [NGEET/fates#855]
There are a number of confusing carbon cycle variables that are output for FATES-SP cases including mortality fluxes, vegetation C pools, and even heterotrophic respiration fluxes.

Can we remove these from the FATES-sp output?

**Comment by adrifoster** [comment]
I agree, I think we should try to pare down the SP-mode output to only include variables that are relevant in SP mode.

@rgknox @glemieux we had discussed refactoring the variable output... would this require a huge refactoring as well?

**Comment by ekluzek** [comment]
The easiest way to start this would be to create a user-mod directory for FATES-SP (which wouldn't be bad to do anyway). It could then exclude history fields that aren't useful, and/or include only the ones that make sense. Later you might want to turn the defaults on and off according to the fates SP control item.

**Comment by ckoven** [comment]
agreed that this is confusing.  I see a few ways to do this. The simplest way (I think) is to make a character variable, something like `true_if_not_sp_mode` that is either `active` or `inactive`, depending on the configuration, and use that as the argument to `use_default` in `initialize_history_vars`.   Then a user could still override if they wanted.

 A more complex way would be to wrap the variable definitions in logic like we already do here: https://github.com/NGEET/fates/blob/master/main/FatesHistoryInterfaceMod.F90#L4523.  In this case a user couldn't override because the variable simply would't exist.

@ekluzek I'm not sure I understand what you are suggesting, could you describe in greter detail what you mean?

**Comment by glemieux** [comment]
I like the idea of the former @ckoven.  It seems that would be a more flexible way to implement future active/inactive defaults for other modes (if desired).

**Comment by ekluzek** [comment]
@ckoven what you suggest is the more robust longer term solution. What I suggest to start with though is that you setup a user-mod directory under cime_config/usermod_dirs, for fates_sp. It would then have a user_nl_clm file that would first turn FATES_SP on, but then also have hist_fincl_excl1 for the list of things that should be excluded.

You then add "--user-mods-dirs fates_sp" to your create_newcase when you are running. The user-mod could also be used for testing.

One disadvantage is that it's specific for CTSM. But, it gives you a chance to figure out the complete list of fields that do and don't make sense for fates-sp. And it's easy to add a new field if needed. Since, FATES-SP is a major mode for running FATES it makes sense that it have at least a user-mod directory to turn it on. Later we want to make turning it on more similar to how you turn on big leaf SP mode.

**Comment by ckoven** [comment]
got it, thanks @ekluzek.  I guess the usermods approach would be compatible with the first approach that I described as well, so either or both could be done in any order.  For the second approach that I described, it wouldn't be compatible with the usermods approach because (I think) if a `hist_fincl_excl1` points to a variable that hasn't been defined, then it would crash, right?  So if there is anything where we really don't think it can or should be defined in a given mode (as in the case I linked to) and want to not allow it even as a non-default option, then we should probably just do that and avoid any future conflicts.

**Comment by ekluzek** [comment]
Hi @ckoven, the hist_fincl_excl will work provided the history field is at least defined. It doesn't matter if it's default on or off it'll still work. So they do totally work together. You are right that if you have an if statement around a hist_addfld call, so it isn't even defined it would error out.

**Comment by ckoven** [comment]
I just did a first runthrough of variables that, based on my (subjective) perspective, are either not defined or aren't really very meaningful in SP mode, and followed the (first) approach that I described above to make them non-default.  It's actually a pretty long list of variables that we don't want default in SP mode.  So I think maybe we do want to try to keep this in the fortran, just to keep all the info in one place.  I suspect that the list of default variables also needs some general going over anyway to pare down the more esoteric ones.  But there are also most likely variables outside the FATES scope (soil carbon, etc) that we also would want to make non-default in FATES-SP mode, and we should definitely use @ekluzek's approach to do that as well.

commit is here: https://github.com/ckoven/fates/commit/cc3101a

If people agree both that (a) this way of handling things makes the most sense, and (b) this list of variables looks right, then I can add this either as its own PR or onto another PR that I will be submitting soon that has a few other minor changes and bugfixes to history variables.

**Comment by wwieder** [comment]
Hi Everyone,

It seems like some of this is also wrapped up in #859. 
I have an additional question that's related to the upcoming CTSM tutorial (May 25-26).  Currently a sample of the FATESsp output that students will generate and be plotting is available here `/glade/scratch/wwieder/archive/I2000_CTSM_FATESsp/lnd/hist/`. 

Are we OK to provide history files with these extra variables, or is this something we'd like to correct before the tutorial?

**Comment by wwieder** [comment]
great, here are some other FATES-side variables we can consider removing in FATES-sp history output.  Maybe @ckoven or @adrifoster can confirm?

FATES_AUTORESP, FATES_GROWTH_RESP
FATES_LEAFC, FATES_LEAFC_PF, FATES_LEAFMAINTAR, FATES_MAINT_RESP,
FATES_NPP, FATES_NPP_PF,
FATES_c_to_litr_cel_c, FATES_c_to_litr_lab_c, FATES_c_to_litr_lig_c,

**Comment by ckoven** [comment]
@wwieder I think that `FATES_LEAFC` and `FATES_LEAFC_PF` are meaningful in SP mode.  I.e. not prognostic, but still meaningful, as in for sanity-checking purposes.  The other ones you mention could be added to the do-not-include list.

**Comment by chengyun19** [comment]
> 我同意，我认为我们应该尝试减少 SP 模式输出以仅包含与 SP 模式相关的变量。
> 
> @rgknox @glemieux我们已经讨论过重构变量输出……这是否也需要大量重构？

Hello,
I am a novice at running FATES. I find that the default SP model is false, I want to simulate GPP and NPP form 1970 to 2010. Do I need to turn on SP? And when should we turn on SP? I am not sure, can you help me? And in the I2000CLM50FATES compset, the CO2 is constant. Does this mean that I need to input the historical 1970-2010 CO2 data to get the 1970-2010 GPP and NPP?
Any suggestions will be appreciated! Thanks in advance.

**Comment by adrifoster** [comment]
Check to see if this is still relevant
